jetty-core/jetty-util/src/main/java/org/eclipse/jetty/util/UrlEncoded.java

```diff
@@ -329,8 +329,7 @@ private static void decodeUtf8To(String query, int offset, int length, BiConsume
             switch (c)
             {
                 case '&':
-                    value = buffer.toReplacedString();
-                    buffer.reset();
+                    value = buffer.takeString(IllegalArgumentException::new);
```

Comment:
Message-less Exceptions are horrible

Reply:
Agree. Specially for a generic exception type like IAE. CharacterCodingException doesn't take a message, but the name kind of says it.  Might need to ressurect NotUtf8Exception